### PR TITLE
Use fallback user avatar in cases where the user is unknown to us

### PR DESCRIPTION
**Pull request by Kicu**
### Details

 - the logic that decides which avatar to show is now heavily consolidated into the `<Avatar>` component
 - right now the source of truth is almost always `avatar` prop from `personalDetails`. It is returned from backend, and any user "known" to us will have such `accountId`
 - we will now display `FallbackAvatar` when:
   - there is no details/`accountID`
   - when `accountID` was optimistically set in TS code
   - as a final fallback in Avatar component (that behaviour is unchanged)
 - I eliminated 30+ calls to `getAvatar()` because what they mostly did was use `source` + `accountID` to decide whether to show SVG version of default avatar or Concierge avatar. Now `source` prop is simply passed through the components and the call to `getAvatar()` is done only once inside `<Avatar />`.

#### special note for icons:
 - I tried to avoid manually setting `FallbackAvatar` but sometimes when creating `icons` object I have to do it, because of patterns like this one: https://github.com/Expensify/App/blob/main/src/components/OptionRow.tsx#L207 - The existence of `icons` object decides whether any kind of Avatar will be shown. That is why you will find `?? FallbackAvatar` in some places

### Fixed Issues

$ https://github.com/Expensify/App/issues/38743
PROPOSAL:

### Tests

1. go to every place in the app where we can display an unknown user and verify that the avatar next to them is the fallback one (and not the colorful one)
2. places to verify:
 - search option from Chats list
 - searching users from "Start chat"
 - searching users from "Request money"
 - searching users from "Send money"
 - searching users from "Assign Task"
 - searching users from Workspace list > Workspace > Members > Invite member 
3. browse through the app and verify that no other avatar logic was broken

### QA Steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/3929868/e135737d-81d5-4063-985c-c3cbc13e1bf6

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/3929868/561fdace-95f1-4652-9f03-e8be74fecb07

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/3929868/49d8d80f-395c-4c7f-8680-87cad6aedec7

https://github.com/Expensify/App/assets/3929868/9ec5c174-d81f-4592-97d1-4e36cfb0f633

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Review by blazejkustra — APPROVED** [review]

File: src/pages/DetailsPage.tsx
Comment (s77rt) [inline src/pages/DetailsPage.tsx:73]:
```suggestion
```

**Review by s77rt — CHANGES_REQUESTED** [review]
Avoid explicitly setting the fallback avatar i.e. replace all instance were we return the fallback avatar with returning null or undefined. Then in the Avatar component if no source is provided then use the fallback avatar

**Comment by Kicu** [comment]
@s77rt that is a good comment in general and I'd prefer to not return the fallback avatar explicitly.
However originally I didn't do it, since there are places that use `icons` and they require a source, which means because of typescript I need to go over multiple places in the app and allow a missing avatar/icon to pass through so that in the end the fallback can be applied.
This might require a lot of changes, but perhaps its worth it.

I will try to clean this up so that we never pass `FallbackAvatar` through function calls

**Comment by Kicu** [comment]
@s77rt I have now spent a few hours trying to refactor to returning undefined instead of fallback avatar.
That will require changes in a lot of places, because the function `getAvatar` (https://github.com/Expensify/App/blob/main/src/libs/UserUtils.ts#L158) would have to start returning `AvatarSource | undefined`.

That function is used in 10+ files multiple times, and it affects many places - multiple parts of the app that display avatars depend on it. Often its used when creating `icons[]` which are of type `OnyxCommon.Icon` which requires a source to always be there. I don't want to touch logic for displaying multiple icons or multiple avatars.

Some examples of usage that you can check:
 - usage when creating several icons: https://github.com/Expensify/App/blob/main/src/libs/ReportUtils.ts#L1608
 - as part of icons for 1 item: https://github.com/Expensify/App/blob/main/src/pages/home/report/ReactionList/BaseReactionList.tsx#L75

In general this change would affect so many places in the code that I believe its not worth doing in this PR.
I suggest that we keep my original code. What do you think?

File: src/libs/OptionsListUtils.ts
Comment (s77rt) [inline src/libs/OptionsListUtils.ts:353]: Revert. We will be using detail.avatar which is from onyx.

**Review by s77rt — CHANGES_REQUESTED** [review]
Found 2 missing cases where we should use the fallback avatar if user is not found in onyx

- In ReportUtils.getPersonalDetailsForAccountID
- In ProfilePage

**Comment by s77rt** [comment]
We may have a problem in `ReportUtils.getDisplayNamesWithTooltips` because it has a param `personalDetailsList` that may accept data that is both from or not from onyx. It's not clear if the avatar should use default or fallback.

Currently using default which causes the following bug (in money request)
Screenshot 2024-04-05 at 1 31 50 AM

**Comment by Kicu** [comment]
@s77rt 
Thanks for pointing out the problem. That bug made me go down the rabbit hole of the avatar logic and I spend a lot of time trying come up with a reasonable fix.
... At this point I don't believe any sensible fix is possible 😅

**tl;dr**
We are generating optimistic IDs for participants before we even call backend and we do this in several places of the app. Because of it there is no easy way to guess whether a `participantDetails` object represents an actual user that we "know" or if it represents just temporary optimistic data create somewhere in the app.


Example places where this happens (there might be more):
 - `getOptions` (https://github.com/Expensify/App/blob/main/src/libs/OptionsListUtils.ts#L1815-L1819) which is used in multiple places by `<OptionsList` or similar components; any callbacks passed down to options list will get personal details lists, and there might be optimistic id inside.
 For example in here: https://github.com/Expensify/App/blob/main/src/pages/iou/steps/MoneyRequstParticipantsPage/MoneyRequestParticipantsSelector.js#L168 - its very difficult to know whether accountID inside participant was optimistic or not
 - `getAccountIDsByLogins` (https://github.com/Expensify/App/blob/main/src/libs/PersonalDetailsUtils.ts#L99-L101) - used when opening new report and set as `optimisticData` in api call inside `Report.openReport()`. This gets saved to Onyx as optimistic data.
 Again, when outside of context of calling `Report.openReport()` there are components which do not have knowledge whether an `accountID` is optimistic or not

As you can see in the described examples ☝️ I think there is no way to simply tell whether the user IS or IS NOT known to us, and base the fallback avatar logic on this.
Calling @grgia since you originally created this issue.

At this point Im not sure what is the best way forward, I'm reluctant to modify code related to optimistic accountID generation since it touches multiple places in the app.
Or perhaps I misunderstood something related to this task, in that case point me in a better direction.

**Comment by grgia** [comment]
@Kicu 

We can definitely call it on this one if it's proving a rabbit hole to fix. It's not high enough priority if it's proven to be complex.

One idea to fix this is by tracing/removing all of the `getDefaultAvatar` calls in app, if we just store the default avatar in the source for users (BE) then we'd be able to tell which users we know/have an avatar set. Then default to the fallback whenever the source is empty. 

We'd then use one single call to `getDefaultAvatar` in the avatar component to switch to the crisp SVGs/grab the local concierge, etc. avatars. 

Let me know if that's something you already ruled out

**Comment by Kicu** [comment]
@grgia Thanks for the suggestion, I think this is something that is worth investigating.
> if we just store the default avatar in the source for users (BE) then we'd be able to tell which users we know/have an avatar set.

In this scenario would do the storing? Frontend code when making an api call and sending it to backend? or backend would do this on their end and always return an avatar url?

I'm asking because this is related to the "is the user known to us" issue - Im no longer sure I know when we decide that the user "is known" to us 😅 is it when b

Otherwise I think this is a sound idea to base the avatar on 1 field (url) instead of calling `getDefaultAvatar` + `getAvatar` in 20+ places in the app as we do right now.

The only complication I can see is that in some places we don't use a property `avatar` but rather create an array of `icons` (example: https://github.com/Expensify/App/blob/main/src/libs/OptionsListUtils.ts#L373-L380 or here: https://github.com/Expensify/App/blob/main/src/libs/OptionsListUtils.ts#L1838).
I worry if that no icons will be returned then that might affect some display logic. 
But that is something I will try to quickly investigate now.

Tell me what you think

**Review by blazejkustra — CHANGES_REQUESTED** [review]
Code looks much cleaner now, but you have conflicts and a lint failing 👍

File: src/components/Avatar.tsx
Comment (Kicu) [inline src/components/Avatar.tsx:93]: this line gives us 100% guarantee that there will be something in the `source` prop, so that we will at the worst always display `FallbackAvatar` if both `source` and `accountID` are missing.

Thanks to that we can just pass props directly into the `Avatar` component, and we don't have to call `.getAvatar()` in multiple places in code.

**Comment by s77rt** [comment]
@Kicu Can you please let me know once this is ready for review?

**Comment by Kicu** [comment]
@s77rt this PR is now fully updated and ready for a re-review. 

The whole solution changed and I was able to remove almost all calls to `getAvatar()` which I believe greatly simplifies the logic.
The description, tests and video recordings are all updated. Please take a look when you have time :)

File: src/pages/home/report/ReportActionItemSingle.tsx
Comment (s77rt) [inline src/pages/home/report/ReportActionItemSingle.tsx:104]:
```suggestion
        avatarAccountId = action.delegateAccountID;
```
Already a number

File: src/pages/home/report/ReportActionItemSingle.tsx
Comment (s77rt) [inline src/pages/home/report/ReportActionItemSingle.tsx:136]:
```suggestion
        id: avatarAccountId,
```
Update the `avatarAccountId` in the if condition `isWorkspaceActor`

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by s77rt** [comment]
Missing icon for optimistic user (probably similar to  https://github.com/Expensify/App/pull/39229#discussion_r1566194298 https://github.com/Expensify/App/pull/39229#discussion_r1566279895)



https://github.com/Expensify/App/assets/16493223/0aefe1b4-86c5-41f1-8b73-d1a64a1d2a0b

File: src/components/ReportActionItem/TaskView.tsx
Comment (Kicu) [reply src/components/ReportActionItem/TaskView.tsx:159]: This is fine, undefined will not happen because this is wrapped in a check `report.managerID ? (...` in line 155.

File: src/libs/OptionsListUtils.ts
Comment (Kicu) [reply src/libs/OptionsListUtils.ts:315]: I think this is a leftover.
Originally I thought that it makes no sense to create `Icon` when there is no source, so I did a lot of this `? []` pattern. Only later I noticed that sometimes the icons will be missing if there is nothing to show. I missed this case.
You are 100% correct about this causing regressions. Will fix

File: src/pages/home/report/ReactionList/BaseReactionList.tsx
Comment (Kicu) [reply src/pages/home/report/ReactionList/BaseReactionList.tsx:66]: will fix

**Comment by Kicu** [comment]
@s77rt thanks for the review. I fixed almost all places that you pointed out, and I double checked that I always create `icons` object, because this is the source of many regressions.

Both of the bugs you found are now working correctly:
<p>
Screenshot 2024-04-16 at 11 28 39
Screenshot 2024-04-16 at 11 31 14
</p>

File: src/components/Avatar.tsx
Comment (Kicu) [reply src/components/Avatar.tsx]: I did moreless what you suggested here, however I added an extra `isWorkspace ?` check before running `UserUtils`, since the Avatar component can handle both types of avatars.

It felt unsafe to just always run `getAvatar()` on the original source.

File: src/pages/home/report/ReportActionItemSingle.tsx
Comment (Kicu) [reply src/pages/home/report/ReportActionItemSingle.tsx]: fixed

File: src/pages/home/report/ReportActionItemSingle.tsx
Comment (Kicu) [reply src/pages/home/report/ReportActionItemSingle.tsx]: what I wanted to avoid is passing the `avatarAccountId` because I thought that for a "workspaceActor" there is some other id in this variable. 
But I guess you are right so I fixed this.

File: src/pages/home/report/ReportActionItemSingle.tsx
Comment (s77rt) [inline src/pages/home/report/ReportActionItemSingle.tsx:136]: You missed to set `avatarAccountId` to `undefined` if `isWorkspaceActor`

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by s77rt** [comment]
@Kicu for future PRs please avoid force push as it messes up the review process and I don't get to know what I reviewed vs. what I still have to review

**Comment by s77rt** [comment]
## Reviewer Checklist

- [X] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [X] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


android

</details>

<details>
<summary>Android: mWeb Chrome</summary>


mweb-chrome

</details>

<details>
<summary>iOS: Native</summary>


ios

</details>

<details>
<summary>iOS: mWeb Safari</summary>


mweb-safari

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


web

</details>

<details>
<summary>MacOS: Desktop</summary>


desktop

</details>

**Comment by s77rt** [comment]
When creating a chat the tooltip on the name is different than that on the avatar

https://github.com/Expensify/App/assets/16493223/4e568cea-41a2-4b37-8c06-956917c3cc09

**Comment by s77rt** [comment]
When adding a member the avatar in the details page have wrong avatar

https://github.com/Expensify/App/assets/16493223/66d79668-5fd6-41f4-bc05-79f080736937

File: src/pages/home/report/ReportActionItemSingle.tsx
Comment (Kicu) [reply src/pages/home/report/ReportActionItemSingle.tsx:136]: @s77rt but last time you told me to change it here to this specific code: https://github.com/Expensify/App/pull/39229#discussion_r1566306188

so which version do you think it should be?

Comment (s77rt) [reply src/pages/home/report/ReportActionItemSingle.tsx:136]: I meant to set that here
https://github.com/Expensify/App/blob/445a70e923134e4a19444e84b61e595fff0b6af2/src/pages/home/report/ReportActionItemSingle.tsx#L92-L96

below `avatarSource = ...` add `avatarAccountId = undefined`

Comment (Kicu) [reply src/pages/home/report/ReportActionItemSingle.tsx:136]: okay understood 👍

**Comment by Kicu** [comment]
@s77rt Thanks for being patient with this PR :) I pushed a new commit where I fixed 2 things:
 - the `undefined` in `ReportActionItemSingle`
 - this comment https://github.com/Expensify/App/pull/39229#issuecomment-2062632286 (wrong tooltip after chat creation) - the extra check inside `Avatar` will now handle this.

As for your last comment this is tricky to fix.
Until you click on the member we have no data about this user and the fallback avatar correctly displays. However when we click on him a call to backend is made which returns user data **including** an avatar. Then since we have the avatar - we show it. I hope my video can explain this better:

https://github.com/Expensify/App/assets/3929868/4e99ae8a-ddb7-470a-bddc-b0e39fe718df



So I think this behavior should stay as is. Please re-review.
(and I will stop force pushing 👍 )

File: src/components/Avatar.tsx
Comment (s77rt) [inline src/components/Avatar.tsx:90]: Revert. There shouldn't be a case where originalSource is empty and source is not. The bug is in `UserUtils.isDefaultAvatar`. Remove this part

https://github.com/Expensify/App/blob/95001698b2c3173f8c3d043458282e8e75bf90ce/src/libs/UserUtils.ts#L143-L146

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by s77rt** [comment]
This bug https://github.com/Expensify/App/pull/39229#issuecomment-2062634001 is on staging so let's not block on that

File: src/components/Avatar.tsx
Comment (Kicu) [reply src/components/Avatar.tsx]: Done 👍  `UserUtils.isDefaultAvatar` is actually used in 2 more places, but I checked both and this change shouldn't break anything.
Uploading avatar displays correct options, and QR Share also works.

**Comment by Kicu** [comment]
@s77rt ready for re-review

**Review by s77rt — APPROVED** [review]

File: src/pages/ProfilePage.tsx
Comment (blazejkustra) [inline src/pages/ProfilePage.tsx:99]: Nit: You can delete this line and use `details.avatar` inside the component

**Review by blazejkustra — APPROVED** [review]
LGTM!

File: src/components/Avatar.tsx
Comment (grgia) [inline src/components/Avatar.tsx:59]: what's the originalSource?

**Review by grgia — COMMENTED** [review]
Code / changes LGTM, quick question before merging:

Are we still displaying the SVG files?

**Comment by s77rt** [comment]
Yes https://github.com/Expensify/App/pull/39229#discussion_r1577790352

**Review by grgia — APPROVED** [review]

**Comment by grgia** [comment]
Thanks @s77rt !

**Pull request merged**
